resources/adoptopenjdk_install.rb

```diff
@@ -0,0 +1,178 @@
+#
+# Cookbook:: java
+# Resource:: adoptopenjdk_install
+#
+# Based on oracle_install.rb by Bryan W. Berry (<bryan.berry@gmail.com>)
+#
+
+resource_name :adoptopenjdk_install
+
+default_action :install
+
+property :url, String
+property :checksum, String, regex: /^[0-9a-f]{32}$|^[a-zA-Z0-9]{40,64}$/
+property :md5, String, regex: /^[0-9a-f]{32}$|^[a-zA-Z0-9]{40,64}$/
+property :app_home, String
+property :app_home_mode, Integer, default: 0755
+property :bin_cmds, Array, default: []
+property :owner, String, default: 'root'
+property :group, String, default: lazy { node['root_group'] }
+property :default, [true, false], default: true
+property :alternatives_priority, Integer, default: 1
+property :reset_alternatives, [true, false], default: true
+property :variant, ['hotspot', 'openj9', 'openj9-large-heap'], default: 'openj9'
+
+action :install do
+  app_dir_name, tarball_name = parse_app_dir_name(new_resource.url)
+  app_root = new_resource.app_home.split('/')[0..-2].join('/')
+  app_dir = app_root + '/' + app_dir_name + '-' + new_resource.variant
```

Comment:
better (and faster) to use ruby string interpolation over string concatenation as this requires creating multiple strings instead of a single string. Also same comment here about the duplication.